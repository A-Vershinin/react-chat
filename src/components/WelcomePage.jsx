import React, { Component, Fragment } from 'react';
import { Redirect } from 'react-router-dom';
import { withStyles } from '@material-ui/core/styles';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Grid from '@material-ui/core/Grid';
import Paper from '@material-ui/core/Paper';
import Tabs from '@material-ui/core/Tabs';
import Tab from '@material-ui/core/Tab';
import Typography from '@material-ui/core/Typography';
import LoginForm from './LoginForm.jsx';
import SignUpForm from './SignUpForm.jsx';
import ErrorMessage from './ErrorMessage.jsx';

const styles = theme => ({
  paper: {
    // eslint-disable-next-line
    marginTop: 64 + theme.spacing.unit * 3,
    width: 500,
  },
  tabContent: {
    padding: theme.spacing.unit * 3,
  },
});

class WelcomePage extends Component {
  state = {
    activeTab: 0,
  };

  handleTabChange = (event, value) => {
    this.setState({ activeTab: value });
  };

  render() {
    const {
      classes, login, signup, isAuthenticated, error,
    } = this.props;
    const { activeTab } = this.state;

    if (isAuthenticated) {
      return <Redirect to="/chat" />;
    }

    return (
      <Fragment>
        <AppBar>
          <Toolbar>
            <Typography variant="title" color="inherit" style={{ flex: 1 }}>
              DogeCodes React Chat
            </Typography>
          </Toolbar>
        </AppBar>
        <Grid container justify="center">
          <Grid item>
            <Paper className={classes.paper}>
              <AppBar position="static" color="default">
                <Tabs value={activeTab} fullWidth onChange={this.handleTabChange}>
                  <Tab label="Login" />
                  <Tab label="Sign Up" />
                </Tabs>
              </AppBar>
              <div className={classes.tabContent}>
                {activeTab === 0 && <LoginForm onSubmit={login} />}
                {activeTab === 1 && <SignUpForm onSubmit={signup} />}
              </div>
            </Paper>
          </Grid>
        </Grid>
        <ErrorMessage error={error} />
      </Fragment>
    );
  }
}

export default withStyles(styles)(WelcomePage);
