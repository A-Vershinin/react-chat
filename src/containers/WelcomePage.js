import { connect } from 'react-redux';
import WelcomePage from '../components/WelcomePage.jsx';
import { signupAction, loginAction } from '../actions';

const mapStateToProps = state => ({
  isAuthenticated: state.auth.isAuthenticated,
  error: state.services.errors.auth,
});

const mapDispatchToProps = dispatch => ({
  signup: (username, password) => dispatch(signupAction(username, password)),
  login: (username, password) => dispatch(loginAction(username, password)),
});

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(WelcomePage);
