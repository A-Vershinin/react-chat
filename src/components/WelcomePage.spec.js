/* eslint-env jest */
import React from 'react';
import ReactDOM from 'react-dom';
import { MemoryRouter, Route } from 'react-router-dom';
import renderer from 'react-test-renderer';
import WelcomePage from './WelcomePage.jsx';

jest.mock('./LoginForm', () => () => 'LoginForm');
jest.mock('./SignUpForm', () => () => 'SignUpForm');
jest.mock('./ErrorMessage', () => () => 'ErrorMessage');

const mockProps = {
  classes: {
    paper: 'some text a',
    tabContent: 'some text b',
  },
  login: jest.fn(),
  signup: jest.fn(),
  isAuthenticated: false,
  error: null,
};

describe('<WelcomePage />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <MemoryRouter initialEntries={['/']}>
        <Route path="/welcome" render={props => <WelcomePage {...mockProps} {...props} />} />
      </MemoryRouter>,
      div,
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders correctly', () => {
    const tree = renderer
      .create(<MemoryRouter initialEntries={['/']}>
          <Route path="/welcome" render={props => <WelcomePage {...mockProps} {...props} />} />
              </MemoryRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders when user is isAuthenticated', () => {
    const tree = renderer
      .create(<MemoryRouter initialEntries={['/']}>
          <Route path="/welcome" render={props => <WelcomePage {...mockProps} {...props} />} />
              </MemoryRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
