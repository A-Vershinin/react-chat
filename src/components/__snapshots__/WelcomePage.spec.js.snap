// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WelcomePage /> renders correctly 1`] = `null`;

exports[`<WelcomePage /> renders when user is isAuthenticated 1`] = `null`;
